pandas/tests/frame/test_analytics.py

```diff
@@ -1805,6 +1805,15 @@ def test_built_in_round(self):
             {'col1': [1., 2., 3.], 'col2': [1., 2., 3.]})
         tm.assert_frame_equal(round(df), expected_rounded)
 
+    def test_round_nonunique_categorical(self):
+        # See GH21809
+        idx = pd.CategoricalIndex(['low'] * 3 + ['hi'] * 3)
+        dfb = pd.DataFrame(np.random.rand(6, 3), columns=list('abc'),
+                           index=idx)
+        assert dfb.shape == (6, 3)
```

Comment:
can you construct the result and use tm.assert_frame_equal here

Reply:
Added a `tm.assert_frame_equal()` check but keeping the explicit dims check as I don't want to inadvertently rely on `CategoricalIndex.reindex`.